src/storage/mod.rs

```diff
@@ -1396,6 +1398,49 @@ impl<E: Engine, L: LockManager> Storage<E, L> {
                 .await?
         }
     }
+
+    /// Update max_ts with the start_ts and check the given range for locks
+    /// which blocks the reading.
+    pub fn check_memory_locks_in_ranges(
+        &self,
+        mut ctx: Context,
+        start_ts: TimeStamp,
+        ranges: Vec<KeyRange>,
+    ) -> impl Future<Output = Result<Option<LockInfo>>> {
+        let concurrency_manager = self.concurrency_manager.clone();
+        self.read_pool
+            .spawn_handle(
```

Comment:
Because this operation might take some time, it's not appropriate to run it directly in the grpc thread.
I think the read pool is semantically correct for it and the read pool also runs this check for other read commands.